lightning/src/ln/channelmanager.rs

```diff
@@ -2660,34 +2710,32 @@ where
 		let _persistence_guard = PersistenceNotifierGuard::notify_on_drop(
 			&self.total_consistency_lock, &self.persistence_notifier,
 		);
-		{
-			let mut channel_state_lock = self.channel_state.lock().unwrap();
-			let channel_state = &mut *channel_state_lock;
-			for channel_id in channel_ids {
-				let channel_counterparty_node_id = channel_state.by_id.get(channel_id)
-					.ok_or(APIError::ChannelUnavailable {
-						err: format!("Channel with ID {} was not found", log_bytes!(*channel_id)),
-					})?
-					.get_counterparty_node_id();
-				if channel_counterparty_node_id != *counterparty_node_id {
-					return Err(APIError::APIMisuseError {
-						err: "counterparty node id mismatch".to_owned(),
-					});
-				}
+		let per_peer_state = self.per_peer_state.read().unwrap();
+		let peer_state_mutex_opt = per_peer_state.get(counterparty_node_id);
+		if let None = peer_state_mutex_opt {
+			return Err(APIError::APIMisuseError{ err: format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id) });
+		}
+		let mut peer_state_lock = peer_state_mutex_opt.unwrap().lock().unwrap();
```

Comment:
Something like this would be more idiomatic/concise: 
```suggestion
		let peer_state_mutex = per_peer_state.get(counterparty_node_id)
		    .ok_or_else(|| APIError::APIMisuseError{ err: format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id) })?;
		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
```
and elsewhere